llvm/lib/Transforms/Vectorize/VPlan.cpp

```diff
@@ -1546,6 +1540,34 @@ void VPSlotTracker::assignNames(const VPBasicBlock *VPBB) {
       assignName(Def);
 }
 
+std::string VPSlotTracker::getName(const VPValue *V) {
+  auto *UV = V->getUnderlyingValue();
+  auto *VPI = dyn_cast_or_null<VPInstruction>(V->getDefiningRecipe());
+  if (!UV)
+    return VPI->getName().str();
+
+  std::string Name;
+  raw_string_ostream S(Name);
+  if (MST) {
```

Comment:
Might be simpler to handle all cases that don't require constructing a new MST first, with early exit
```suggestion
  if (MST || UV->hasName() || !isa<Instruction>(UV)) {
    UV->printAsOperand(S, false);
    return Name;
  } 
  
  // Lazily create the ModuleSlotTracker when we first hit an unnamed
  // instruction
  auto *IUV = cast<Instruction>(UV);
  // This check is required to support unit tests with incomplete IR.
  if (IUV->getParent()) {
    MST = std::make_unique<ModuleSlotTracker>(IUV->getModule());
    MST->incorporateFunction(*IUV->getFunction());
  } else {
    MST = std::make_unique<ModuleSlotTracker>(nullptr);
  }
  UV->printAsOperand(S, false, *MST);
  return Name;
```